packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -278,15 +290,14 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     };
   }
 
-  function resetContextStack() {
-    // Reset the stack
-    reset();
-    // Reset the cursors
-    resetLegacyContext();
-    resetHostContainer();
-
-    // TODO: Unify new context implementation with other stacks
-    resetProviderStack();
+  function resetStack() {
+    if (nextUnitOfWork !== null) {
+      let interruptedWork = nextUnitOfWork.return;
+      while (interruptedWork !== null) {
+        unwindInterruptedWork(interruptedWork);
+        interruptedWork = interruptedWork.return;
+      }
```

Comment:
Should we (DEV-only) assert that the stack is empty after unwinding?

Reply:
Added